docs/txp.md

```diff
@@ -0,0 +1,486 @@
+```
+
+There are two cases check fails (and transaction rejected):
+
+* Minimal fee is negative or is greater than maximal number of coins in the system
+
+   ```minFee < 0 || minFee > maxBound @Coin```
+   - It probably indicates there is a mistake in currently adopted fee
+   policy
+* Actual fee is less than the minimal fee
+
+   ```txFee < minFee```
+
+### GState modification
+
+If all checks above pass, we modify GState appropriately. We modify
+stakes and UTXO.
+
+#### UTXO modification
+
+UTXO modification is trivial:
+
+* All `TxInUtxo` inputs are deleted from UTXO
+* For each output `out` with index `i` we put `(TxInUtxo (hash tx) i,
+  out)` into UTXO.
+
+Note, that `TxInUtxo` is only input type supported in CSL v1.0. For unknown inputs GState is not modified.
+
+#### Computing stakes for transaction output
+
+Each `txOut` can be converted to a list of `(StakeholderId, Coin)` pairs.
+
+This is being done with considering `AddrStakeDistribution` attribute of address corresponding to `txOut`:
+
+- For `BootstrapEraDistr` the value `val` of `txOut` is distributed among
+bootstrap stakeholders proportional to their weights.
+
+    + If `val >= bootDustThreshold`, then each stakeholder receives
+`val * w_ᵢ / weights_sum` coins
+      - one stakeholder also receives the remainder (the choice is deterministc)
+    + Otherwise:
+      - some stakeholders will receive the same stake as their weights
+      - one stakeholder may receive less then their weight
+      - other stakeholders won't receive anything
+- For `SingleKeyDistr id` distribution, the value of `TxOut` will be assigned to stakeholder `id`.
+- For `MultiKeyDistr portions` stake will be distributed among multiple
+stakeholders according to specified `portions`
+     + The first stakeholder may receive slightly more than others due to rounding.
+
+#### Stakes modification
+
+Stakes modification is a bit more complex than UTXO modification, but is also very intuitive:
+
+1. Each `TxInUtxo` input corresponds to some `TxOut`, we compute stakes for it
+   * For all inputs, stake lists are concatenated into one, `inStakes`
+   * This list denotes how much stake each stakeholder should lose
+2. Each output is `TxOut`, we compute stakes for it
+   * For all inputs, stake lists are concatenated into one, `outStakes`
+   * This list denotes how much stake each stakeholder should gain
+3. Stakes of all mentioned stakeholders are updated appropriately
+   * Decreased for all `TxInUtxo` inputs
+   * Increased for all outputs
+4. Total stake is updated
+
+## Local transaction processing
+
+Local transaction processing is needed for transaction relay. Node can
+receive a standalone transaction from the network and then it needs to
+verify it against its current state (global + mempool) and:
+
+* Apply and relay further (to neighbors)
+* Reject
+
+There are few differences between local txp and global txp:
+
+* In local txp we consider not only GState, but also mempool. We
```

Comment:
>In local txp we consider not only GState, but also mempool.

I am not sure this phrase is absolutely clear for a reader because each transaction is part of mempool, I would write **we consider mempool for the moment when tx comes** or smth like that.

Reply:
Disagree. Here we state that GState is accompanied by smth else, that's it

Reply:
> I am not sure this phrase is absolutely clear for a reader because each transaction is part of mempool, I would write we consider mempool for the moment when tx comes or smth like that.

Imo, everything is clear. What is incorrect way to interpret it? 🤔